docs/Modules/abiSessionValidationModule.mdx

```diff
@@ -0,0 +1,559 @@
+        }
+
+        if (callValue > permittedValueLimit) {
+            revert("ABISV Permitted Value Exceeded");
+        }
+
+        // avoided explicit check that (_sessionKeyData.length - 62) is the multiple of RULE_LENGTH
+        // also avoided calculating the rules list length from the rules list itself
+        // both to save on gas
+        // there is a test case that demonstrates that if the incorrect rules list length is provided
+        // the validation will fail
+        if (
+            !_checkRulesForPermission(
+                _funcCallData,
+                rulesListLength,
+                bytes(_sessionKeyData[62:]) //the rest of the _sessionKeyData is the rules list
+            )
+        ) {
+            revert("ABISV Arg Rule Violated");
+        }
+
+        return sessionKey;
+    }
+```
+
+**Operational Flow:**
+
+The method internally calls **_validateSessionParams** to perform the validation against the session permissions.
+It passes the necessary parameters to **_validateSessionParams**  and returns the result.
+
+**_validateSessionParams** validates the parameters of the call against the permissions defined for the session key.
+
+**Execution flow in **_validateSessionParams****
+
+- Extracts essential details from _sessionKeyData, such as session key address, permitted destination contract, permitted method selector, value limit, and rules list length.
+- Checks if the destinationContract matches the permitted destination contract, if the function selector in _funcCallData matches the permitted selector, and if the callValue does not exceed the permitted value limit.
+- Calls **_checkRulesForPermission** to verify if the call data complies with the permission rules.
+
+**Execution flow in **_checkRulesForPermission****
+
+- Iterates through the list of rules.
+- Parses each rule to determine the offset, condition, and value.
+- Verifies if the call data arguments satisfy conditions defined by the rules.
+
+## Rules
+
+Rules define permissions for the args of an allowed method.
+With Rules you can precisely define what should be the args of the transaction that is allowed for a given Session.
+Every Rule works with a single static arg or a 32 bytes chunk of the dynamic arg.
+
+Since the ABI Encoding translates every static param into a 32bytes word, even the shorter ones (like `address` or `uint8`), every Rule defines a desired relation (`Condition`) between n-th 32bytes word of the `calldata` and a reference Value (that is obviously a 32bytes word as well).
+
+So, when dApp is creating a `_sessionKeyData` to enable a session, it should convert every shorter static arg to a 32bytes word to match how it will be actually ABI encoded in the `userOp.callData`.
+
+For the dynamic args, like `bytes`, every 32bytes word of the `calldata` such as offset of the bytes arg, length of the bytes arg, and n-th 32bytes word of the bytes arg can be controlled by a dedicated Rule.
```

Comment:
```suggestion
## Rules

Rules define permissions for the args of an allowed method.
With rules, you can precisely define what should be the args of the transaction that is allowed for a given Session.
Every Rule works with a single static arg or a 32-byte chunk of the dynamic arg.

Since the ABI Encoding translates every static param into a 32bytes word, even the shorter ones (like `address` or `uint8`), every Rule defines a desired relation (`Condition`) between n-th 32bytes word of the `calldata` and a reference Value (that is obviously a 32 bytes word as well).

So, when dApp is creating a `_sessionKeyData` to enable a session, it should convert every shorter static arg to a 32bytes word to match how it will be actually ABI encoded in the `userOp.callData`.

For the dynamic args, like `bytes`, every 32 bytes word of the `calldata` such as offset of the bytes arg, length of the bytes arg, and n-th 32 bytes word of the bytes arg can be controlled by a dedicated Rule.
```